### PR TITLE
Log stats on neo connections

### PR DESCRIPTION
**Pull request by gavin-norman-sociomantic**

### CODE REVIEW

File: src/swarm/node/model/INodeInfo.d [hunk 1]
```diff
@@ -78,6 +78,40 @@ public interface INodeInfo
...
+
+    public size_t num_neo_connections ( );
+
+
+    /**************************************************************************
+
+        Returns:
+            the number of active neo connections currently being handled
+            (Note: the number of idle connections in the pool can be calculated
+            by num_connections - num_open_connections)
```
Comment (nemanja-boric-sociomantic) [inline src/swarm/node/model/INodeInfo.d]: `by num_neo_connections - num_open_neo_connections`

File: src/swarm/node/model/Node.d [hunk 2]
```diff
@@ -241,6 +241,21 @@ public abstract class INodeBase : INode, INodeInfo
...
+    /**************************************************************************
+
+        Returns:
+            the limit of the number of neo connections (i.e. the maximum number
+            of connections the node can handle in parallel) or 0 if limitation
+            is disabled
+
+     **************************************************************************/
+
+    override public size_t neo_connection_limit ( )
```
Comment (nemanja-boric-sociomantic) [inline src/swarm/node/model/Node.d:253]: Bah, this is useless, could we do something about this? Maybe make the `INodeInfo` abstract class and provide default implementation? I'm happy to leave it as is and wait for better days for the swarm's node framework.

Comment (gavin-norman-sociomantic) [reply src/swarm/node/model/Node.d:253]: Puh... yeah... I wanted to mess with the horrible inheritance hierarchy as little as possible in this PR.

File: src/swarm/node/model/INodeInfo.d (hunk 1 above)
Comment (gavin-norman-sociomantic) [reply src/swarm/node/model/INodeInfo.d]: Updated.

**Review by nemanja-boric-sociomantic — APPROVED** [review]

**Pull request merged**